core/crop/structure.py

```diff
@@ -78,17 +80,19 @@ class SensorClass(BASE):
     __tablename__ = SENSOR_TABLE_NAME
 
     # columns
-    id = Column(Integer, primary_key=True, autoincrement=True)
+    id = Column(Integer, primary_key=True)
     type_id = Column(
         Integer,
         ForeignKey("{}.{}".format(SENSOR_TYPE_TABLE_NAME, ID_COL_NAME)),
         nullable=False,
     )
     device_id = Column(Unicode(100), nullable=False)
+    installation_date = Column(DateTime, nullable=False)
 
     time_created = Column(DateTime(), server_default=func.now())
     time_updated = Column(DateTime(), onupdate=func.now())
 
+    #FIXME:One sensor to many loc?
```

Comment:
Yes, but not at the same time. SensorLocationClass has the restriction: __table_args__ = (UniqueConstraint("sensor_id", "installation_date"),)

Reply:
ah. got it ! good point